src/atopile/front_end.py

```diff
@@ -792,32 +793,61 @@ def _init_node(self, stmt_ctx: ParserRuleContext, ref: Ref) -> L.Node:
             for super_ctx in promised_supers:
                 self.visitBlock(super_ctx.block())
 
-        new_node.add_trait(from_dsl(stmt_ctx))
+        new_node.add(from_dsl(stmt_ctx))
         return new_node
 
+    def _get_or_promise_param(
```

Comment:
This is practically splitting this and the function below it into discrete functions based on their intent